File: 04web_iot/03btn_notify.py
from flask import Flask, request, redirect
from flask_socketio import SocketIO, send
import RPi.GPIO as GPIO

app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret'
socketio = SocketIO(app,async_mode='threading')

GPIO.setmode(GPIO.BCM)
pin_btn = 23

GPIO.setup(pin_btn, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)

@app.route('/')
def main():
    return redirect('/static/btn_notify.html')


@socketio.on('connect')
def connect():
    send({'data': 'welcome'})
    
def my_callback(channel):
    print 'Edge detected on channel %s state %s'% (channel, GPIO.input(channel)) 
    val =GPIO.input(channel)
    if val == 1:
        socketio.send( {'data': '1' })
    else:
        socketio.send({'data': '0'})
    print 'btn pressed', 0


if __name__ == '__main__':
    try:
        GPIO.add_event_detect(pin_btn, GPIO.BOTH, callback=my_callback)
        socketio.run(app, host='0.0.0.0')
    finally:
        print 'cleaning up'
        GPIO.cleanup()

File: 04web_iot/03btn_notify.js
var http = require('http');
var express = require('express');
var socketio = require('socket.io');
var onoff = require('onoff');

var app = express();
var server = http.createServer(app);
var io = socketio(server);
var sw_pin = new onoff.Gpio(23, 'in', 'both');

sw_pin.watch(function(err, value){
	console.log('btn: '+ value);
	io.emit('message', {data: value});
});

app.use(express.static(__dirname + '/static'));
app.get('/', function(req, res){
	res.redirect('/btn_notify.html');
});

io.on('connection', function(socket){
	console.log('connected' + socket.id);
	 socket.emit('message',{'data': 'welcome'});
});

server.listen(5000, function(){
	console.log('server running on 5000.');
});